wp/lib/bap_wp/src/z3_utils.ml

```diff
@@ -111,3 +111,59 @@ let mk_smtlib2_single (env : Env.t) (smt_post : string) : Constr.t =
   let ctx = Env.get_context env in
   mk_smtlib2 ctx smt_post decl_syms
 
+let construct_pointer_constraint (l: string list) (env1 : Env.t)
+    (env2: Env.t option): Constr.t =
+  let stack_end = Env.get_stack_end env1 in
+  let ctx = Env.get_context env1 in
+  let arch = match Env.get_arch env1 |> Bap.Std.Arch.addr_size with
+    | `r32 -> 32
+    | `r64 -> 64 in
+  let rsp = Env.get_rsp_name env1 in
+  let err_msg_input = "invalid register name in pointer register list" in
+  let err_msg_rsp = "stack pointer not found" in
+  let sb_bv = Z3.BitVector.mk_numeral ctx (stack_end |> Int.to_string) arch in
+  let gen_constr = match env2 with
+    | Some env2 ->
+      let init_var_map_orig = Env.get_init_var_map env1 in
+      let init_var_map_mod = Env.get_init_var_map env2 in
+      (* we do want exceptions here if the register names are invalid
+       *  or RSP doesn't exist *)
+      let rsp_orig = Option.value_exn ~message:("original " ^ err_msg_rsp)
+          (get_z3_name init_var_map_orig rsp Var.name) in
+      let rsp_mod = Option.value_exn ~message:("modified " ^ err_msg_rsp)
+          (get_z3_name init_var_map_mod rsp Var.name) in
+      (* Encode constraint that each register is not within stack*)
+      (fun acc reg ->
+         let reg_name_orig = Option.value_exn ~message:err_msg_input
+             (get_z3_name init_var_map_orig reg Var.name) in
+         let reg_name_mod = Option.value_exn ~message:err_msg_input
+             (get_z3_name init_var_map_mod reg Var.name) in
+         (* the pointer register must be above RSP
```

Comment:
As a side note, if you can get your hands on a module of type `CPU`, you can use this function: https://binaryanalysisplatform.github.io/bap/api/odoc/bap/Bap/Std/module-type-CPU/index.html#val-sp to get your hands on the actual variable that contains the top of the stack.

Reply:
Yes! I added this into `environment.ml` as `get_rsp_name`. I realize the name is x86 specific, so I'm going to change the name.